### PR TITLE
Mulitple memory server - services initialization

### PR DESCRIPTION
**Pull request by faizan-barmawer**
- All services read options from the configuration file.
  - Added get_config_info methods in *_Direct classes to read
    options from config file
  - Initialization sequence for various combination of services.

All combination of different services needs to be tested.

### CODE REVIEW

**Comment by rajakrishi** [comment]
Can one of the admins verify this patch?

File: src/cis/fam_cis_direct.cpp [hunk 1]
```diff
@@ -873,4 +919,61 @@ void Fam_CIS_Direct::get_addr(void *memServerFabricAddr,
...
+
+        try {
+            std::vector<std::string> temp =
+                info->get_value_list("metadata_list");
+            ostringstream metasrvList;
+
+            for (auto item : temp)
+                metasrvList << item << ",";
+
+            options["metadata_list"] = metasrvList.str().c_str();
```
Comment (mashoodk) [inline src/cis/fam_cis_direct.cpp:970]: Clarification : There is a parameter, rpc_interface, how are we incorporating that ? Or you think we need not keep it.

**Review by mashoodk — CHANGES_REQUESTED** [review]

**Comment by faizan-barmawer** [comment]
Addressed review comments from Mashood and updated a new commit.

Regards,
Faizan

**Comment by faizan-barmawer** [comment]
Addressed additional review comments. 
Retained only one build for memoryserver model. Added test runs for different service configuration in build_and_test script.

Regards,
Faizan

File: build_and_test.sh [hunk 2]
```diff
@@ -1,105 +1,144 @@
 #!/bin/bash
 CURRENTDIR=`pwd`
-CIS_RPC_BUILD_DIR=$CURRENTDIR/build/build-cis-rpc
+RPC_BUILD_DIR=$CURRENTDIR/build/build-rpc
```
Comment (gautham-bhat-k) [inline build_and_test.sh:3]: Will it be appropriate to name build-mem-server or similar name ?

File: build_and_test.sh [hunk 3]
```diff
@@ -1,105 +1,144 @@
...
 SHM_BUILD_DIR=$CURRENTDIR/build/build-shm
-ALL_RPC_BUILD_DIR=$CURRENTDIR/build/build-all-rpc
 
 #creating build directory if not present
-if [ ! -d "$CIS_RPC_BUILD_DIR" ]; then
-  echo "Creating $CIS_RPC_BUILD_DIR"
-  mkdir -p $CIS_RPC_BUILD_DIR
+if [ ! -d "$RPC_BUILD_DIR" ]; then
```
Comment (gautham-bhat-k) [inline build_and_test.sh:7]: Instead of RPC_BUILD_DIR can it be MEMSERV_BUILD_DIR ?

**Review by gautham-bhat-k — CHANGES_REQUESTED** [review]

File: build_and_test.sh (hunk 2 above)
Comment (faizan-barmawer) [reply build_and_test.sh]: Renamed the build directory for memory server model and pushed a new commit.

Regards,
Faizan

**Review by mashoodk — APPROVED** [review]

**Pull request merged**